lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -1176,18 +1161,23 @@ bool SessionState::init_new_monitor(
 
 void SessionState::update_session_level_key(
     const UsageMonitoringUpdateResponse& update,
-    SessionStateUpdateCriteria& update_criteria) {
+    SessionStateUpdateCriteria& uc) {
   const auto& new_key = update.credit().monitoring_key();
-  if (session_level_key_ != nullptr && *session_level_key_ != new_key) {
-    MLOG(MWARNING) << "Session level monitoring key already exists, updating";
+  if (session_level_key_ != "" && session_level_key_ != new_key) {
+    MLOG(MINFO) << "Session level monitoring key is updated from "
+                << session_level_key_ << " to " << new_key;
   }
   if (update.credit().action() == UsageMonitoringCredit::DISABLE) {
-    session_level_key_ = nullptr;
-    // TODO: set in UpdateCriteria
+    session_level_key_ = "";
```

Comment:
offtopic: i haven't gone through this code yet, but we can only have one session_level_key active at the same time, is that correct?

Reply:
Yes, I believe that is the case!